.github/workflows/full-build.yml

```diff
@@ -237,96 +227,52 @@ jobs:
-        id: output
-        run: Write-Output "cachekey=${{ runner.os }}-gvsbuild-${{ env.gvsbuild_update }}-${{ env.gvsbuild_version }}" >> $env:GITHUB_OUTPUT
-      - name: Upload libraries
-        if: steps.cache.outputs.cache-hit != 'true'
-        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
-        with:
-          name: gtk-build
-          path: C:\gtk-build\gtk\x64\release
-
   windows:
     name: Windows
-    needs: [lint, windows-build-gtk]
+    needs: lint
     runs-on: windows-latest
     env:
-      INCLUDE: C:\gtk-build\gtk\x64\release\include;C:\gtk-build\gtk\x64\release\include\cairo;C:\gtk-build\gtk\x64\release\include\glib-2.0;C:\gtk-build\gtk\x64\release\include\gobject-introspection-1.0;C:\gtk-build\gtk\x64\release\lib\glib-2.0\include;
-      LIB: C:\gtk-build\gtk\x64\release\lib
+      INCLUDE: C:\gtk\include;C:\gtk\include\cairo;C:\gtk\include\glib-2.0;C:\gtk\include\gobject-introspection-1.0;C:\gtk\lib\glib-2.0\include;
+      LIB: C:\gtk\lib
+      gvsbuild_version: 2024.5.0
     permissions:
       contents: write
     timeout-minutes: 45
     outputs:
       installer: gaphor-${{ steps.install.outputs.version }}-installer.exe
     if: "!contains(github.event.head_commit.message, 'skip ci')"
     steps:
-      - name: Harden Runner
-        uses: step-security/harden-runner@a4aa98b93cab29d9b1101a6143fb8bce00e2eac4 # v2.7.1
-        with:
-          egress-policy: audit
-
-      - name: Create GTK binaries dir
-        run: mkdir C:\gtk-build\gtk\x64\release
-      - name: Get GTK binaries from cache
-        uses: actions/cache/restore@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      - name: GTK binaries get from cache
+        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
         id: cache
         with:
-          path: C:\gtk-build\gtk\x64\release\**
-          key: ${{ needs.windows-build-gtk.outputs.cachekey }}
-      - name: Check cache hit
+          path: C:\gtk\**
+          key: ${{ runner.os }}-gvsbuild-${{ env.gvsbuild_version }}
+      - name: Download Gvsbuild
         if: steps.cache.outputs.cache-hit != 'true'
+        env:
+          REPO: "wingtk/gvsbuild"
+          MATCH: "^GTK4_Gvsbuild_.+_x64.zip$"
+          RENAME: "Gvsbuild.zip"
+        shell: bash
         run: |
-          echo "No GTK binaries cache"
-          exit 1
+          curl -sL --fail \
+            -H "Accept: application/vnd.github.v3+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+              "https://api.github.com/repos/${REPO}/releases/tags/${{ env.gvsbuild_version }}" \
+              | jq -r ".assets | .[] | select(.name | test(\"${MATCH}\")) | .url" \
+              | tee asset.url
+          curl -sL --fail \
+            -H "Accept: application/octet-stream" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+              -o "${RENAME}" \
+              "$(cat asset.url)"
```

Comment:
Wouldn't it be nicer if we can use the Github CLI here?

Reply:
Nice suggestion, I updated to use the GitHub CLI.